contracts/access/IAccessControlDefaultAdminRules.sol

```diff
@@ -11,6 +11,39 @@ import "./IAccessControl.sol";
  * _Available since v4.9._
  */
 interface IAccessControlDefaultAdminRules is IAccessControl {
+    /**
+     * @dev The new default admin is not a valid default admin.
+     */
+    error AccessControlInvalidDefaultAdmin(address defaultAdmin);
+
+    /**
+     * @dev Role can't be granted.
+     */
+    error AccessControlForbiddenGrant(bytes32 role);
+
+    /**
+     * @dev Role can't be revoked.
+     */
+    error AccessControlForbiddenRevoke(bytes32 role);
```

Comment:
It's correct, these two errors are only used in this contract. However, I planned this error to be as generic as possible with the intention of avoiding multiple errors for the same purpose.

I think this is according to the idea of reusing the errors wherever it makes sense. I think we care about making these reusable. A counterexample would be a use case in which revoking the (eg.) `EXECUTOR` role is forbidden if there's no other `EXECUTOR` left.

I'd be more in favor of moving these errors to `AccessControl` instead, but didn't do it because they aren't used there.

I don't think we should remove the role, but I'm open to the possibility.

Reply:
Lets leave it like this. 

I would not put it in `AccessControl`, because that would add it to the ABI of contracts that don't emit emit. It won't be an issue unless we build another AccessControl based contract, that does not inherit ACDAR, and that needs the error. (that is the case that I think a library would support better, but its not something we are facing now)